oca/recipe/odoo/runtime/session.py

```diff
@@ -228,29 +222,28 @@ def clean_environments(self, reinit=True):
         :param bool reinit: if ``True``, :meth:`init_environments` will be
                             called again after cleaning
         """
-        try:
+        if version_info[0] < 15:
             gen_context = self._environments_gen_context
-        except AttributeError:
-            return
-
-        try:
-            next(gen_context)
-        except StopIteration:
-            pass
-        else:
-            logger.warn("clean_environments: we had the context manager, but "
-                        "it had not been called. This suggest low-leve "
-                        "tampering with it that should be more cautious. "
-                        "Proceeding with cleansing.")
             try:
                 next(gen_context)
             except StopIteration:
                 pass
             else:
-                raise RuntimeError("Called the environments context manager "
-                                   "twice and it's not finished. "
-                                   "This is really unexpected.")
-        del self._environments_gen_context
+                logger.warn(
+                    "clean_environments: we had the context manager, but "
+                    "it had not been called. This suggest low-leve "
```

Comment:
```suggestion
                    "it had not been called. This suggest low-level "
```

Reply:
when we touch this, let's correct the typo

Reply:
Thanks, taken!